components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsDialogFragment.kt

```diff
@@ -178,6 +185,7 @@ internal class SitePermissionsDialogFragment : AppCompatDialogFragment() {
                 putString(KEY_TITLE, title)
                 putInt(KEY_TITLE_ICON, titleIcon)
                 putBoolean(KEY_SHOULD_INCLUDE_CHECKBOX, shouldIncludeCheckBox)
+                putBoolean(KEY_SHOULD_SELECT_REMEMBER_CHOICE, shouldSelectRememberChoice)
                 putBoolean(KEY_USER_CHECK_BOX, shouldIncludeCheckBox)
```

Comment:
We have to update the`KEY_USER_CHECK_BOX (userSelectionCheckBox)` with the value of `shouldSelectRememberChoice` because this is the property that stores the user decision. Right now it's using `shouldIncludeCheckBox` value if the user hits the positive button his/her decision will be store even thought the checkbox is not unchecked.

```kotlin
        positiveButton.setOnClickListener {
            feature?.onPositiveButtonPress(sessionId, userSelectionCheckBox)
            dismiss()
      }
```

Reply:
`shouldIncludeCheckBox`  only gets updated when the user hit checkbox.
  
```kotlin
  containerView.findViewById<CheckBox>(R.id.do_not_ask_again).apply {
            visibility = VISIBLE
            isChecked = checked
            setOnCheckedChangeListener { _, isChecked ->
                userSelectionCheckBox = isChecked
            }
        }
```

Reply:
We don't need anymore `KEY_USER_CHECK_BOX`